src/librbd/object_map/UpdateRequest.cc

```diff
@@ -72,6 +72,15 @@ template <typename I>
 void UpdateRequest<I>::handle_update_object_map(int r) {
   ldout(m_image_ctx.cct, 20) << "r=" << r << dendl;
 
+  if (r == -ENOENT && m_ignore_enoent) {
+    // ignore potential race with deleting a snapshot when copyup
+    // races with a snapshot removal
```

Comment:
nit: I would prefer to see this comment in CopyupRequest, wher the update object map request is called. Because that is where I would be wondering why it was called with ignore enoent set for that particular case. Additionally, this comment might become not exactly true if we have yet another case where we want to ignore enoent.